### ISSUE TITLE
Preboot 6.0.0-beta.1 doesn't seem to do anything when used with Angular Upgrade (ngUpgrade)

### ISSUE THREAD

**Issue by mgol** [angular/preboot#67]
**Note: for support questions, please use the Universal Slack Channcel or https://gitter.im/angular/universal** 

* **I'm submitting a ...**
- [x] bug report
- [ ] feature request

* **Which parts of preboot are affected by this issue?**
- [ ] server side
- [x] client side
- [ ] inline
- [ ] build process
- [ ] docs
- [ ] tests

* **Do you want to request a *feature* or report a *bug*?**

A bug. There was already a question about that above in the issue template...

* **What is the current behavior?**

Preboot doesn't seem to do anything when hooked up as described in README in an app using Angular Upgrade. Input focus is not preserved, its value is lost. I've also noticed `AppComponent#ngOnInit` is fired twice.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** by creating a github repo.

1. Clone https://github.com/mgol/angular-preboot-bug and follow README instructions to run the app.
2. Open Chrome DevTools and enable network throttling, setting it to "Slow 3G".
3. Load the page and when the input appears, write something in it.
4. Wait for the JS to finish downloading and Angular to kick in.

* **What is the expected behavior?**

Input focus state & its value should be preserved.

* **What is the motivation / use case for changing the behavior?**

Core functionality of Preboot doesn't work due to this bug.

* **Please tell us about your environment:**

- Browser: Chrome 64
- Language: TypeScript 2.6.2
- OS:  macOS
- Platform: Node.js

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

If you disable Angular Upgrade (in the linked reproduction repo you can revert to the previous commit on `master`) `AppComponent#ngOnInit` is fired only once but Preboot still doesn't work due to #66.

**Comment by CaerusKaru** [comment]
Noting that this is a feature request that likely won't make the fray until v6 is out of beta.

**Comment by mgol** [comment]
tl;dr: jump to the code at the end to see the workaround.

The fact that `AppComponent#ngOnInit` fires twice is caused by buffering - during the buffering phase there are two elements named `app-root` on the page and with ngUpgrade it's AngularJS that's responsible for finding the element and initializing them (those elements are Angular components downgraded to AngularJS via ngUpgrade). Disabling the buffering makes `AppComponent#ngOnInit` fired only once but the events are still not replayed correctly (I assume they're firing too early). You can workaround that, too, by disabling automatic event replaying and doing that yourself in `AppComponent#ngOnInit`. You need to wrap that in a `setTimeout`, though, or it will happen too early.

The drawback is that while manual replaying seems to work well, disabling the buffering has a visible effect as the screen shortly flickers.

The full workaround: in `AppModule` change the import `PrebootModule.withConfig({appRoot: 'app-root'})` to:
```ts
    PrebootModule.withConfig({
      appRoot: 'app-root',
      buffer: false,
      noReplay: true,
    }),
```
In `AppComponent#ngOnInit` replay events manually in a `setTimeout`:
```ts
  ngOnInit() {
    this.router.initialNavigation();
    setTimeout(() => {
      this.replayer.replayAll();
    });
  }
```

**Comment by mgol** [comment]
A partial fix: #68. That PR fixes the buffering issue so `buffer: true` works. It still requires manual event replaying as in the above workaround but works otherwise.
